src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -1,408 +0,0 @@
-                if (lastResult.IsDefault)
-                    return LoadInitialAnalysisData(project, version, cancellationToken);
-
-                RoslynDebug.Assert(lastResult.DocumentIds != null);
-
-                // PERF: avoid loading data if version is not right one.
-                // avoid loading data flag is there as a strictly perf optimization.
-                if (lastResult.Version != version)
-                {
-                    return lastResult;
-                }
-
-                // if given project doesnt have any diagnostics, return empty.
-                if (lastResult.IsEmpty)
-                {
-                    return DiagnosticAnalysisResult.CreateEmpty(lastResult.ProjectId, lastResult.Version);
-                }
-
-                // loading data can be canceled any time.
-                var serializerVersion = lastResult.Version;
-                var builder = new Builder(project, lastResult.Version, lastResult.DocumentIds);
-
-                foreach (var documentId in lastResult.DocumentIds)
-                {
-                    cancellationToken.ThrowIfCancellationRequested();
-
-                    var document = project.GetDocument(documentId);
-                    if (document == null)
-                        continue;
-
-                    if (!TryGetDiagnosticsFromInMemoryStorage(serializerVersion, document, builder))
-                    {
-                        Debug.Assert(lastResult.Version == VersionStamp.Default);
-
-                        // this can happen if we merged back active file diagnostics back to project state but
-                        // project state didn't have diagnostics for the file yet. (since project state was staled)
-                        continue;
-                    }
-                }
-
-                if (!TryGetProjectDiagnosticsFromInMemoryStorage(serializerVersion, project, builder))
-                {
-                    // this can happen if SaveAsync is not yet called but active file merge happened. one of case is if user did build before the very first
-                    // analysis happened.
-                }
-
-                return builder.ToResult();
-            }
-
-            /// <summary>
-            /// Return all diagnostics for the given document stored in this state including non local diagnostics for this document
-            /// </summary>
-            public async Task<DiagnosticAnalysisResult> GetAnalysisDataAsync(TextDocument document, CancellationToken cancellationToken)
-            {
-                // make a copy of last result.
-                var lastResult = _lastResult;
-                Contract.ThrowIfFalse(lastResult.ProjectId == document.Project.Id);
-
-                var version = await GetDiagnosticVersionAsync(document.Project, cancellationToken).ConfigureAwait(false);
-                if (lastResult.IsDefault)
-                    return LoadInitialAnalysisData(document, version);
```

Comment:
lastResult.IsDefault would be true practically always. meaning we would virtually always go into LoadInitialAnalysisData.